.github/workflows/linux.yml

```diff
@@ -11,12 +12,22 @@ jobs:
         with:
           path: src/github.com/aws/amazon-ecs-agent
       - name: get GO_VERSION
+        id: get-go-version
         run:  |
           cd $GITHUB_WORKSPACE/src/github.com/aws/amazon-ecs-agent
-          echo "GO_VERSION=$(cat GO_VERSION)" >> $GITHUB_ENV
+          set -eou pipefail
+          go_version=$(cat -e GO_VERSION)
+          go_version=${go_version%?}
+          go_version_length=${#go_version}
+          go_version_re="^([0-9]+\.){2}([0-9]+)$"
```

Comment:
This might fail to match the first major version releases for example `1.18`. Using `^([0-9]+\.){1,2}([0-9]+)$` is probably safer